File: ai-captioner/backend/services/audioService.js
const ffmpeg = require('fluent-ffmpeg');
const ffmpegPath = require('ffmpeg-static');

ffmpeg.setFfmpegPath(ffmpegPath);

/**
 * 동영상에서 오디오를 추출합니다.
 */
function extractAudio(videoPath, audioPath) {
    return new Promise((resolve, reject) => {
        ffmpeg(videoPath)
            .toFormat('wav')
            .audioCodec('pcm_s16le')
            .audioFrequency(16000)
            .audioChannels(1)
            .on('end', () => {
                console.log('Audio extraction finished');
                resolve();
            })
            .on('error', (err) => {
                console.error('Error during audio extraction:', err);
                reject(err);
            })
            .save(audioPath);
    });
}

module.exports = { extractAudio };

File: ai-captioner/backend/services/audio_service.py
import subprocess
import os

class AudioService:
    @staticmethod
    def extract_audio(video_path: str, output_path: str):
        """
        동영상 파일에서 오디오를 추출하여 mp3/wav로 저장합니다.
        FFmpeg이 설치되어 있어야 합니다.
        """
        print(f"Extracting audio from {video_path} to {output_path}...")
        try:
            ffmpeg_path = r"C:\MagicMic\x86\rtaivc\env\ffmpeg.exe"
            command = [
                ffmpeg_path,
                '-i', video_path,
                '-vn', # 비디오 제외
                '-acodec', 'libmp3lame',
                '-y', # 덮어쓰기 허용
                output_path
            ]
            subprocess.run(command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
            return True
        except subprocess.CalledProcessError as e:
            print(f"FFmpeg error: {e.stderr.decode()}")
            return False
        except Exception as e:
            print(f"Error during audio extraction: {e}")
            return False

audio_service = AudioService()
